content/client-server-api/modules/receipts.md

```diff
@@ -87,6 +122,84 @@ not have their notification counts rewound to that point in time. While
 uncommon, it is considered valid to have an `m.read` (public) receipt lag
 several messages behind the `m.read.private` receipt, for example.
 
+##### Threaded read receipts
+
+{{% added-in v="1.4" %}}
+
+If a client does not use [threading](#threading), then they will simply only
+send "unthreaded" read receipts which affect the whole room regardless of threads.
+
+Threading introduces a concept of multiple conversations being held in the same
+room and thus deserve their own read receipts and notification counts. An event
+is considered to be "in a thread" if it has either a `rel_type` of `m.thread` or
+has child events with a `rel_type` of `m.thread` (in which case it'd be the
+thread root). Events not in a thread but still in the room are considered to be
+part of the "main timeline", or a special thread with an ID of `main`.
```

Comment:
Should this have similar language [to above](https://github.com/matrix-org/matrix-spec/pull/1255/files#diff-bfda8c579b08f5dd97e312b0a592762a36737a8477e23dc9273936eb9b2d6aebR126) discussing going 3 hops to find the thread?

Reply:
possibly, though it's not immediately obvious that it's what the MSC intended - will dig a bit deeper on it.

Reply:
Both MSC3771 and MSC3773 have similar language like:

> It is recommended that at least 3 relations are traversed when attempting to find a thread, implementations should be careful to not infinitely recurse.

This is for both ensuring the receipt refers to a valid event ID & when partitioning a push action by thread.